src/psyclone/tests/psyir/nodes/array_of_structures_reference_test.py

```diff
@@ -104,6 +106,22 @@ def test_asr_create(component_symbol):
     assert isinstance(asref.children[1], nodes.Range)
     check_links(asref, asref.children)
     check_links(asref.children[1], asref.children[1].children)
+
+    # Test to enforce a type:
+    lbound = nodes.BinaryOperation.create(
+        nodes.BinaryOperation.Operator.LBOUND,
+        nodes.Reference(component_symbol), int_one.copy())
+    ubound = nodes.BinaryOperation.create(
+        nodes.BinaryOperation.Operator.UBOUND,
+        nodes.Reference(component_symbol), int_one.copy())
+    my_range = nodes.Range.create(lbound, ubound)
+    # pylint: disable=no-member
+    datatype = psyir.LfricRealScalarDataType()
```

Comment:
Nitpicking but can you use a generic datatype for this test.

I would like the src/psyclone/test/psyir/* to be fully tested with src/psyclone/psyir/* instead of importing domain classes.

Reply:
Good point, done.